database/engine/rrdengine.c

```diff
@@ -11,6 +11,22 @@ rrdeng_stats_t global_flushing_pressure_page_deletions = 0;
 
 static unsigned pages_per_extent = MAX_PAGES_PER_EXTENT;
 
+void *dbengine_page_alloc() {
+#ifdef MADV_MERGEABLE
+    return mymmap(NULL, RRDENG_BLOCK_SIZE, MAP_PRIVATE, enable_ksm);
+#else
```

Comment:
@ilyam8 if `mymmap` fails we need to run `fatal()`, because the code does not check.
This is what `mallocz()` does too.